### DISCUSSION TITLE
WalletConnectModal not recognising that metamask is installed

### DISCUSSION THREAD

**Discussion by vitoff34** [WalletConnect/walletconnect-monorepo#3538] [Developer Support]
Hello,
I am building a WalletConnectModal functionality into my app. I want it to detect when my user has installed metamask or other wallets on desktop (in browser).  This happens automatically when using the Web3Modal. This is what I want: 

<img width="379" alt="metamaskinstalled" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/127387534/2f69cfa7-083d-4d1d-86ad-2e86c975421b">

Here is my code for WalletConnectModal: 
```
const modal = new WalletConnectModal({
    projectId: project_id,
    // chains: ['eip155:1', 'eip155:42161', 'eip155:137'],
    chains: ['eip155:1'],
    // enableAuthMode: true
    privacyPolicyUrl: 'https://bet777.io',
    termsOfServiceUrl: 'https://bet777.io',
    explorerRecommendedWalletIds: [
        'c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96', // this is metamask wallet id
        '4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0'
    ]
})
```
However I get this: 

<img width="379" alt="metamasknotinstalled" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/127387534/adf1ddfa-1cef-4154-b4cf-293d1a154e2c">

And when a user clicks on metamask, even if he has it installed in the browser the modal gives a QR code to scan with phone:

<img width="379" alt="metamaskqr" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/127387534/9f733843-e12d-4781-bf1e-590c581eece4">

How can I make the WalletConnectModal detect metamask or other installed wallets on desktop?
Thanks!

**Comment by nordost8** [comment]
I recently set everything up according to your documentation, and auto-detection Metamask/Coinbase extension was working, but now it isn't. Coinbase and MetaMask are being ignored in PC browser
Google Chrome
Version 127.0.6533.72 (Official Build) (64-bit)
:(

`import { Web3AuthNoModal } from "@web3auth/no-modal";
import { OpenloginAdapter } from "@web3auth/openlogin-adapter";
import { WalletConnectModal } from "@walletconnect/modal";
import { WalletConnectV2Adapter } from "@web3auth/wallet-connect-v2-adapter";
import { getWalletConnectV2Settings } from "@web3auth/wallet-connect-v2-adapter";
import { CHAIN_NAMESPACES } from "@web3auth/base";
import { EthereumPrivateKeyProvider } from "@web3auth/ethereum-provider";

const commonChainConfig = {
  chainNamespace: CHAIN_NAMESPACES.EIP155,
  chainId: "0x725", // Playblock Mainnet
  rpcTarget: "https://rpc.playblock.io",
  displayName: "Playblock",
  blockExplorerUrl: "https://explorer.playblock.io",
  ticker: "USDB",
  tickerName: "USD PlayBlock",
};

const WalletConnectProjectId = 'SECRET'; 
const web3AuthNetwork = "sapphire_devnet";

const privateKeyProvider = new EthereumPrivateKeyProvider({ config: { chainConfig: commonChainConfig } });

export const initWeb3Auth = async (): Promise<Web3AuthNoModal> => {
  const web3auth = new Web3AuthNoModal({
    clientId: "SECRET",
    chainConfig: commonChainConfig,
    web3AuthNetwork,
  });

  const openloginAdapter = new OpenloginAdapter({
    privateKeyProvider,
    chainConfig: commonChainConfig,
    web3AuthNetwork,
  });

  web3auth.configureAdapter(openloginAdapter);

  const walletConnectModal = new WalletConnectModal({
    projectId: WalletConnectProjectId,
    explorerRecommendedWalletIds: [
      'c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96',
      'fd20dc426fb37566d803205b19bbc1d4096b248ac04548e3cfb6b3a38bd033aa'
    ],
    themeMode: 'dark',
    explorerExcludedWalletIds: 'ALL',
    enableAuthMode: false
  });

  const defaultWcSettings = await getWalletConnectV2Settings(CHAIN_NAMESPACES.EIP155, [commonChainConfig.chainId], WalletConnectProjectId);

  const walletConnectV2Adapter = new WalletConnectV2Adapter({
    adapterSettings: { qrcodeModal: walletConnectModal, ...defaultWcSettings.adapterSettings },
    loginSettings: { ...defaultWcSettings.loginSettings },
  });
  web3auth.configureAdapter(walletConnectV2Adapter);

  await web3auth.init();

  return web3auth;
};
`
